sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -40,56 +40,94 @@ public typealias CommunicationSignalingErrorHandler = (CommunicationSignalingErr
 /// TrouterTokenRefreshHandler for fetching tokens.
 internal typealias TrouterTokenRefreshHandler = (_ stopSignalingClient: Bool, Error?) -> Void
 
-class CommunicationSignalingClient {
-    private var selfHostedTrouterClient: SelfHostedTrouterClient
+public class CommunicationSignalingClient {
+    private var selfHostedTrouterClient: SelfHostedTrouterClient?
     private var communicationSkypeTokenProvider: CommunicationSkypeTokenProvider
-    private var trouterUrlRegistrar: TrouterUrlRegistrar
+    private var trouterUrlRegistrar: TrouterUrlRegistrar?
     private var logger: ClientLogger
     private var communicationHandlers: [ChatEventId: CommunicationHandler] = [:]
+    private var configuration: RealTimeNotificationConfiguration?
+    private var configApiVersion: String
 
+    // Step 1: Initialize with basic properties
     init(
         communicationSkypeTokenProvider: CommunicationSkypeTokenProvider,
         logger: ClientLogger = ClientLoggers.default(tag: "AzureCommunicationSignalingClient")
     ) throws {
         self.logger = logger
         self.communicationSkypeTokenProvider = communicationSkypeTokenProvider
+        self.configApiVersion = "2024-09-01"
```

Comment:
Why is this being hard coded?

Reply:
Currently, the service side supports the version "2024-09-01", which is why it's hardcoded in the client. If a new version is released on the service side in the future, we can update this value accordingly on the client side to align with the latest version. Feel free to let me know if you have other opinions!